sprint_mision_6/sprint-mission-6.sql

```diff
@@ -0,0 +1,311 @@
+WHERE
+    date = '2025-03-12'
+    AND time >= '00:00:00'
+    AND time < '12:00:00';
+
+-- 10. `pizza_types` 테이블에서 이름에 'Cheese' 혹은 'Chicken'이 포함된 피자 종류를 조회하세요. (대소문자를 구분합니다)
+SELECT
+    *
+FROM
+    pizza_types
+WHERE
+    name LIKE '%Cheese%'
+    OR name LIKE '%Chicken%';
+
+-- # 중급 문제
+-- 1. `order_details` 테이블에서 각 피자(`pizza_id`)별로 주문된 건 수(`order_id`)를 보여주세요.
+SELECT
+    pizza_id,
+    COUNT(*)
+FROM
+    order_details
+GROUP BY
+    pizza_id;
+
+-- 2. `order_details` 테이블에서 각 피자(`pizza_id`)별로 총 주문 수량을 구하세요.
+SELECT
+    pizza_id,
+    SUM(quantity)
+FROM
+    order_details
+GROUP BY
+    pizza_id;
+
+-- 3. `pizzas` 테이블에서 `price`의 크기가 20보다 큰 피자의 종류만 `order_details` 테이블에서 조회하세요. (힌트: 서브쿼리)
+SELECT
+    pizza_id
+FROM
+    order_details
+WHERE
+    pizza_id IN (
+        SELECT
+            id
+        FROM
+            pizzas
+        WHERE
+            price > 20
+    );
+
+-- 4. `orders` 테이블에서 각 날짜별 총 주문 건수를 `order_count` 라는 이름으로 계산하고, 하루 총 주문 건수가 80건 이상인 날짜만 조회한 뒤, 주문 건수가 많은 순서대로 정렬하세요.
+SELECT
+    date,
+    COUNT(*) AS order_count
+FROM
+    orders
+GROUP BY
+    date
+HAVING
+    COUNT(*) >= 80
+ORDER BY
+    order_count DESC;
+
+-- 5. `order_details` 테이블에서 피자별(`pizza_id`) 총 주문 수량이 10개 이상인 피자만 조회하고, 총 주문 수량 기준으로 내림차순 정렬하세요.
+SELECT
+    pizza_id,
+    SUM(quantity) AS pizza_count
+FROM
+    order_details
+GROUP BY
+    pizza_id
+HAVING
+    SUM(quantity) >= 10
+ORDER BY
+    pizza_count DESC;
+
+-- 6. `order_details` 테이블에서 피자별 총 수익을 `total_revenue` 라는 이름으로 구하세요. (수익 = `quantity * price`)
+SELECT
+    o.pizza_id,
+    SUM(o.quantity * p.price) AS total_revenue
+FROM
+    order_details AS o
+    JOIN pizzas AS p ON o.pizza_id = p.id
+GROUP BY
+    o.pizza_id;
+
+-- 7. 날짜별로 피자 주문 건수(`order_count`)와 총 주문 수량(`total_quantity`)을 구하세요.
+SELECT
+    o.date,
+    COUNT(DISTINCT o.id) AS order_count,
+    SUM(d.quantity) AS total_quantity
+FROM
+    orders AS o
+    JOIN order_details AS d ON d.order_id = o.id
+GROUP BY
+    o.date;
+
+-- # 고급 문제
+/*
+ 1. 피자별(`pizzas.id` 기준) 판매 수량 순위에서 피자별 판매 수량 상위에 드는 베스트 피자를 10개를 조회해 주세요. `pizzas`의 모든 컬럼을 조회하면서 각 피자에 해당하는 판매량을 `total_quantity`라는 이름으로 함께 조회합니다.
+ 
+ 
+ 출력 예시:
+ 
+ ```sql
+ big_meat_s    | big_meat    | S    |    12 |           1914
+ thai_ckn_l    | thai_ckn    | L    | 20.75 |           1410
+ five_cheese_l | five_cheese | L    |  18.5 |           1409
+ four_cheese_l | four_cheese | L    | 17.95 |           1316
+ classic_dlx_m | classic_dlx | M    |    16 |           1181
+ spicy_ital_l  | spicy_ital  | L    | 20.75 |           1109
+ hawaiian_s    | hawaiian    | S    |  10.5 |           1020
+ southw_ckn_l  | southw_ckn  | L    | 20.75 |           1016
+ bbq_ckn_l     | bbq_ckn     | L    | 20.75 |            992
+ bbq_ckn_m     | bbq_ckn     | M    | 16.75 |            956
+ ```
+ */
+SELECT
+    p.*,
+    SUM(od.quantity) AS total_quantity
+FROM
+    pizzas AS p
+    JOIN order_details AS od ON p.id = od.pizza_id
+GROUP BY
+    p.id
+ORDER BY
+    total_quantity DESC
+LIMIT
+    10;
```

Comment:
풀스캔을 피하고, PSQL은 SELECT의 비집계 컬럼을 모두 GROUP BY에 포함해야해요

```suggestion
SELECT p.id, p.type_id, p.size, p.price,
       SUM(od.quantity) AS total_quantity
FROM pizzas p
JOIN order_details od ON p.id = od.pizza_id
GROUP BY p.id, p.type_id, p.size, p.price
ORDER BY total_quantity DESC
LIMIT 10;
```